kratos/python/add_utilities_to_python.cpp

```diff
@@ -422,6 +422,32 @@ std::string GetRegisteredNameCondition(const Condition& rCondition)
     return name;
 }
 
+void RecursiveEnsureModelPartOwnsProperties(AuxiliarModelPartUtilities& rAuxiliarModelPartUtilities)
+{
+    rAuxiliarModelPartUtilities.RecursiveEnsureModelPartOwnsProperties();
+}
+
+void RecursiveEnsureModelPartOwnsPropertiesWithClear(
+    AuxiliarModelPartUtilities& rAuxiliarModelPartUtilities,
+    const bool RemovePreviousProperties
+    )
+{
+    rAuxiliarModelPartUtilities.RecursiveEnsureModelPartOwnsProperties(RemovePreviousProperties);
+}
+
+void EnsureModelPartOwnsProperties(AuxiliarModelPartUtilities& rAuxiliarModelPartUtilities)
+{
+    rAuxiliarModelPartUtilities.EnsureModelPartOwnsProperties();
+}
+
+void EnsureModelPartOwnsPropertiesWithClear(
+    AuxiliarModelPartUtilities& rAuxiliarModelPartUtilities,
+    const bool RemovePreviousProperties
+    )
+{
+    rAuxiliarModelPartUtilities.EnsureModelPartOwnsProperties(RemovePreviousProperties);
+}
```

Comment:
As a comment, I think that at some point we should start promoting the use of lambdas for this.

Reply:
Indeed

Reply:
I think I have a better proposal for this: [pybinds `overload_cast`](https://pybind11.readthedocs.io/en/stable/classes.html)

If I understood it correctly then we don't need any lambdas or wrapper functions

Downside: its C++14 :(

I think it is still worth waiting, bcs then we don't have to do the work twice and all of our python-interfaces will become much cleaner :D

Reply:
I wasn't aware of this feature.

In any case, I wouldn't forbid the use of lambdas for this. Of course, not to change whatever it is already there, but at least for the new exported methods.

Reply:
> Of course, not to change whatever it is already there

Thats basically what I meant ;)

Reply:
I added the lambdas, it compiles with C++11, if you don't like it, it can be reverted